### PR TITLE
My idea: Fully worked out

### PR DESCRIPTION
**Pull request by MrKickkiller**
PlaneFactory: Still bears my irl name. Suggest change in later commit.

### CODE REVIEW

**Comment by AEnterprise** [comment]
hmm, not sure if we realy want this. seems to make things a bit overcomplex for what we are making atm

File: src/main/java/kineticrevolution/multiblocks/patterns/PatternFactories/Factories/PlaneFactory.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package kineticrevolution.multiblocks.patterns.PatternFactories.Factories;
+
+import kineticrevolution.multiblocks.patterns.PatternException;
+import kineticrevolution.multiblocks.patterns.PatternFactories.IPatternFactory;
+import kineticrevolution.multiblocks.patterns.Pattern;
+
+/**
+ * Created by Mathieu on 3/07/2015.
```
Comment (AEnterprise) [inline src/main/java/kineticrevolution/multiblocks/patterns/PatternFactories/Factories/PlaneFactory.java]: might wana fix the name here

File: src/main/java/kineticrevolution/multiblocks/patterns/PatternFactories/Factories/PlaneFactory.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+ */
+public class PlaneFactory implements IPatternFactory {
+    @Override
+    public Pattern createPattern(int size, char a) throws PatternException {
+        char[][][] pattern = new char[size][size][1];
```
Comment (AEnterprise) [inline src/main/java/kineticrevolution/multiblocks/patterns/PatternFactories/Factories/PlaneFactory.java]: this only allows squares, could you make it allow for rectangles as well?

File: src/main/java/kineticrevolution/multiblocks/patterns/PatternFactories/PatternFactory.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+import kineticrevolution.multiblocks.patterns.PatternException;
+import kineticrevolution.multiblocks.patterns.PatternFactories.Factories.*;
+
+import java.util.ArrayList;
+
+/**
+ * Created by MrKickkiller on 3/07/2015.
+ */
+public final class PatternFactory {
+    final static ArrayList<IPatternFactory> factories = new ArrayList<>();
```
Comment (AEnterprise) [inline src/main/java/kineticrevolution/multiblocks/patterns/PatternFactories/PatternFactory.java:13]: a list and numbered factories doesn't seem the best approach to me, i think an enum would be better here

**Comment by AEnterprise** [comment]
think we'll have this and the current way of doing it manually to give people the choise of what to use. could you provide a few examples on how exactly it works as well?

**Comment by MrKickkiller** [comment]
Example:
Pattern patt = PatternFactory.createHollowBeam(character to be filled in, list of 3 ints, indicating x, y, z)
<patt will now be a Pattern that has a field of char[][][]>

Pattern patt2 = patt.translation(x,y,z) will result in the return a new Pattern with all elements translated x, y, z. Xyz are all positive. no support for negative delta's.

patt.combine(patt2) will result in a new Pattern that has the 2 patterns merged. patt's chars are prefered over patt's

**Comment by AEnterprise** [comment]
merged

**Pull request closed without merging**